### PR TITLE
Added Keyword Spotting Transformer + SpeechCommands example

### PR DESCRIPTION
**Pull request by SarthakYadav**
This PR adds an example for audio classification on the SpeechCommands dataset using a [Keyword Spotting Transformer](https://www.isca-speech.org/archive/interspeech_2021/berg21_interspeech.html), which is essentially a modified Vision Transformer.

The data pipeline is based entirely on `mlx-data`, and based on the SpeechCommands implementation that was recently added to the [mlx-data repository](https://github.com/ml-explore/mlx-data/commit/ae3431648b8e1594d63175a8f121d9873aeb9daa). Also showcases using `mlx.data.features.mfsc` for extracting mel-scaled spectrograms from audio input.

### CODE REVIEW

File: speechcommands/kwt.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        self.attn = Attention(dim, heads, dropout=dropout)
+        self.norm1 = nn.LayerNorm(dim)
+        self.ff = FeedForward(dim, mlp_dim, dropout=dropout)
+        self.norm2 = nn.LayerNorm(dim)
+
+    def __call__(self, x):
+        x = self.attn(x)
+        x = self.norm1(x)
+        x = self.ff(x)
+        x = self.norm2(x)
```
Comment (awni) [inline speechcommands/kwt.py:56]: Is it intentional not to have the skip connections here?

**Review by awni — COMMENTED** [review]
This is looking great, I sent a few updates. I had a question about the skip connections. O/w I think its pretty much ready to go!

File: speechcommands/kwt.py (hunk 1 above)
Comment (SarthakYadav) [reply speechcommands/kwt.py:56]: No, I missed it. I'll redo it and push a commit soon.

Comment (SarthakYadav) [reply speechcommands/kwt.py]: I've fixed this in the latest commit, although changed README with results from only 10 epochs of training. I'll run full 100 epochs of training in a couple of days and add those results. For now can you just go through if there's anything else that you recommend changing?

**Comment by awni** [comment]
Hi @SarthakYadav it looks great, I think we can merge it and you can update the results when you have them. Will you add your name to our contributor section: https://github.com/ml-explore/mlx-examples/blob/main/ACKNOWLEDGMENTS.md#individual-contributors

**Comment by SarthakYadav** [comment]
@awni Sounds good!

> Will you add your name to our contributor section

Done!

**Review by awni — APPROVED** [review]
🚀

**Pull request merged**